satpy/readers/seviri_l1b_native.py

```diff
@@ -276,15 +276,11 @@ def get_lrec(cols):
 
         return np.dtype(drec)
 
-    def _get_memmap(self):
-        """Get the memory map for the SEVIRI data."""
-        with open(self.filename) as fp:
-            data_dtype = self._get_data_dtype()
-            hdr_size = self.header_type.itemsize
-
-            return np.memmap(fp, dtype=data_dtype,
-                             shape=(self.mda["number_of_lines"],),
-                             offset=hdr_size, mode="r")
+    def _get_array(self):
+        """Get the numpy array for the SEVIRI data."""
+        data_dtype = self._get_data_dtype()
+        hdr_size = self.header_type.itemsize
+        return fromfile(self.filename, dtype=data_dtype, offset=hdr_size, count=self.mda["number_of_lines"])
```

Comment:
So here we are basically switching from `np.memmap` to `np.frombuffer` (used inside `fromfile`). I understand this is violating the lazy reading principle we are following throughout satpy. Is this correct? Can we live with it, or shall we maybe add the mechanism to use `np.memmap` in `fromfile` when handling local data?

Reply:
I haven't been following this PR, but for this particular thing I think I agree with @ameraner. Typically memmap should perform better and gives an array-like object back that can be sliced. Perhaps this getting from the file (memmap or not) could be done inside a `da.map_blocks` function if necessary?

Reply:
Thank you @ameraner and @djhoese for your comment on this. I just pushed a commit which addresses the issue by utilizing ``da.map_blocks``.